### PR TITLE
SDK to support the base use case

### PR DESCRIPTION
**Pull request by dev-kiranbanda**
SaaS SDK to support the following:

- activation workflow
- Subscription operations
  - Activation
  - Change quantity ( per user plan )
  - Change Plan
  - Calls to Metering service API 
 - Unsubscribe

Updates to documentation.

### CODE REVIEW

**Comment by code4clouds** [comment]
@dev-kiranbanda There are some conflict files that have to be resolve to allow merging.

**Comment by dev-kiranbanda** [comment]
> @dev-kiranbanda There are some conflict files that have to be resolve to allow merging.

I see that there were modifications to 5 ReadMe files in master and I pulled them into my branch and resolved the conflicts ( I did review the conflicts, took the appropriate changes, fixed lint issues and checked in ).

File: deployment/Database/README.md [hunk 1]
```diff
@@ -1,40 +1,36 @@
...
 The Provisioning and the Publisher applications use SQL Server database as the data source to store subscriptions and the status, metered dimensions by plans and activity against subscriptions.
 
-### Install Using SQL Scripts
+## Install Using SQL Scripts
 
 The SQL scripts build on top of each other. When setting up your database, do the following.
- 
- 1. Start with a blank database
- 2. Run AMP-DB-1.0.sql
- 3. Run AMP-DB-2.0.sql
```
[the remark this answers is not in the document]
Comment (Indrasena-Bijjala-spektrasystems) [reply deployment/Database/README.md:13]: We have removed the additional tables related to ARM templates and deployment. In the AMP-DB-1.0 script file, we have the required tables.

File: deployment/Database/AMP-DB-2.0.sql [hunk 2]
```diff
@@ -1,528 +0,0 @@
-/*Use the database you have created earlied from Master. example: [AMP-DB]*/
```
[the remark this answers is not in the document]
Comment (Indrasena-Bijjala-spektrasystems) [reply deployment/Database/AMP-DB-2.0.sql:1]: We have removed the additional tables related to ARM templates and deployment. In the AMP-DB-1.0 script file, we have the required tables. So the AMP-DB-2.0.sql was removed. 
We are supporting per seat feature through the column "PerUser" in the Plans table.

File: src/SaaS.SDK.Client.DataAccess/Context/SaasKitContext.cs [hunk 3]
```diff
@@ -99,7 +140,7 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                     .WithMany(p => p.KnownUsers)
                     .HasForeignKey(d => d.RoleId)
                     .OnDelete(DeleteBehavior.ClientSetNull)
-                    .HasConstraintName("FK__KnownUser__RoleI__71D1E811");
+                    .HasConstraintName("FK__KnownUser__RoleI__4F7CD00D");
```
[the remark this answers is not in the document]
Comment (dev-kiranbanda) [reply src/SaaS.SDK.Client.DataAccess/Context/SaasKitContext.cs]: I shall park this change as the SQL that we use to create the constraint has to be updated and I am sure this is not a blocker. Shall create a PR later to address this change

File: deployment/Database/AMP-DB-1.0.sql [hunk 4]
```diff
@@ -1,11 +1,25 @@
---Create a database and execute the script
-
---/****** Object:  Table [dbo].[ApplicationLog]    Script Date: 01/15/2020 4:43:08 PM ******/
```
[the remark this answers is not in the document]
Comment (Indrasena-Bijjala-spektrasystems) [reply deployment/Database/AMP-DB-1.0.sql:3]: Updated as suggested. Please approve the PR.

**Pull request merged**